src/swap/components/SwapAmountInput.tsx

```diff
@@ -121,6 +127,60 @@ export function SwapAmountInput({
     }
   }, [token, setToken]);
 
+  useEffect(() => {
+    let isBalanceLoading = balanceResponse?.isLoading;
+    if (token?.symbol === 'ETH') {
+      isBalanceLoading = ethBalanceResponse?.isLoading;
+    }
+    if (type === 'to' && isLoading?.toTokenBalance !== isBalanceLoading) {
+      setIsLoading({ ...isLoading, toTokenBalance: isBalanceLoading });
+    }
+    if (type === 'from' && isLoading?.fromTokenBalance !== isBalanceLoading) {
+      setIsLoading({ ...isLoading, fromTokenBalance: isBalanceLoading });
+    }
+  }, [
+    balanceResponse?.isLoading,
+    ethBalanceResponse?.isLoading,
+    isLoading,
+    setIsLoading,
+    token,
+    type,
+  ]);
+
+  useEffect(() => {
+    if (
+      token?.symbol !== 'ETH' &&
+      balanceResponse?.error &&
+      error?.error !== balanceResponse?.error?.shortMessage
```

Comment:
Just a thought, but this seems could be a small utility that can be shared between the two `useEffect`
```
token?.symbol !== 'ETH' &&
balanceResponse?.error &&
error?.error !== balanceResponse?.error?.shortMessage
```

Reply:
this logic was moved to `Swap`